feat: initial version

mhdawson opened the pull request:
<!-- Thanks for contributing. To speed up the process of reviewing your pull
request please provide us with the following information: -->

## Summary
<!-- A short explanation of the proposed change -->

## Use Cases
<!-- An explanation of the use cases your change enables -->

## Checklist
<!-- Please confirm the following -->
* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

mhdawson commented:
What this needs before it can work/land is

1. Publishes of the ubi-base-stack which should occur one this PR lands - https://github.com/paketo-community/ubi-base-stack/pull/3
1. An update to https://github.com/paketo-community/ubi-nodejs-extension to use the images published in 1)
1. Releases for https://github.com/paketo-community/ubi-nodejs-extension

ForestEckhardt commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1 @@
+* @paketo-buildpacks/builders-maintainers
```
```suggestion
* @paketo-community/builders-maintainers
```

mhdawson commented on builder.toml:
```diff
@@ -0,0 +1,29 @@
+
+[stack]
+  id = "io.buildpacks.stacks.ubi8"
+  build-image = "docker.io/paketocommunity/build-ubi-base/build:0.0.1"
```
```suggestion
  build-image = "docker.io/paketocommunity/build-ubi-base:0.0.1"
```

mhdawson commented:
Everything needed for this to land is now done -  @paketo-community/builders-maintainers.

I validated that with the latest commit I've added the smoke tests run/pass.

mhdawson commented on builder.toml:
```diff
@@ -0,0 +1,29 @@
+
+[stack]
+  id = "io.buildpacks.stacks.ubi8"
+  build-image = "docker.io/paketocommunity/build-ubi-base:0.0.1"
+  run-image = "index.docker.io/paketocommunity/run-ubi-base:0.0.1"
+  run-image-mirrors = ["gcr.io/paketo-community/run-ubi-base:0.0.1"]
```
```suggestion
  build-image = "docker.io/paketocommunity/build-ubi-base:0.0.1"
  run-image = "index.docker.io/paketocommunity/run-ubi-base:latest"
  run-image-mirrors = []
```

robdimsdale commented on go.mod:
```diff
@@ -0,0 +1,51 @@
+module github.com/paketo-community/builder-ubi-base
+
+go 1.18
```
We should probably start off with `go 1.20` given that `go 1.18` is likely to go out of support shortly.

mhdawson replied:
will update and test.

mhdawson replied:
done

robdimsdale commented on smoke/go_test.go:
```diff
@@ -0,0 +1,77 @@
+package smoke_test
```
Are we actually using this test? It seems like `init_test.go` only invokes the nodejs test.

mhdawson replied:
You are correctly I removed all of the tests other than the Node.js one since the builder initially will only have Node.js support.

I'm a bit surprised go did not complain about it, since it complains about most un-used things. Will remove in next commit

robdimsdale reviewed (commented):
Mostly looks good, thanks. A couple of questions inline.

ForestEckhardt commented:
Additionally I am having trouble running the smoke tests because extensions are still an experimental feature in `pack v0.29.0` is there any work around to get the tests to run so that we can validate that things are functioning properly?

ForestEckhardt commented:
@mhdawson I was able to get experimental mode working during build but I am seeing a panic during `pack builder create` so I am not able to get a successful run of the smoke tests yet

mhdawson commented:
@ForestEckhardt I'll re-run the smoke tests locally to see what is going on and also update the scripts to set the required command to enable experimental options.

mhdawson commented:
@ForestEckhardt figured out that I had a later version of pack in by path and that was overriding what was in scripts/.util/tools.json. Updated that and it should work now. 

I did also discover that it seems to work with version v0.30.0-pre2 but not v0.30.0-pre3. Will open issue in pack repo to discuss.

Also should have addressed all comments from @robdimsdale as well.

mhdawson commented:
Issue in pack repo for problems with v0.30.0-pre3 - https://github.com/buildpacks/pack/issues/1850

ForestEckhardt commented:
Alrighty I have gotten the smoke tests to pass. @robdimsdale I would like to get you approval on this as well and then we can merge away.

robdimsdale reviewed (approved):
This generally looks good. I realized that we'll need to add the `tools.json` to the `.syncignore` file otherwise the pack version will be automatically reset to the latest published version. That's inline.

Let's make that change and also open an issue to remove that entry once the next release of `pack` is published.

mhdawson commented on .github/.syncignore:
```diff
@@ -0,0 +1,2 @@
+CODEOWNERS
+dependabot.yml
```
@paketo-community/builders-maintainers issue and commit pushed so this should be ready to land - @paketo-community/builders-maintainers
